lib/activity.php

```diff
@@ -0,0 +1,222 @@
+<?php
+/**
+ * Copyright (c) 2015 Victor Dubiniuk <victor.dubiniuk@gmail.com>
+ * This file is licensed under the Affero General Public License version 3 or
+ * later.
+ * See the COPYING-README file.
+ */
+
+namespace OCA\Files_Antivirus;
+
+use OC\L10N\Factory;
+use OCP\Activity\IExtension;
+use OCP\IURLGenerator;
+
+class Activity implements IExtension {
+	
+	const APP_NAME = 'files_antivirus';
+
+	/**
+	 * Filter with all app related activities
+	 */
+	const FILTER_AVIR = 'files_antivirus';
+
+	/**
+	 * Activity types known to this extension
+	 */
+	const TYPE_VIRUS_DETECTED = 'virus_detected';
+
+	/**
+	 * Subject keys for translation of the subjections
+	 */
+	const SUBJECT_VIRUS_DETECTED = 'virus_detected';
+	
+	/**
+	 * Infected file deletion notice
+	 */
+	const MESSAGE_FILE_DELETED = 'files_deleted';
+
+
+	/** @var Factory */
+	protected $languageFactory;
+
+	/** @var IURLGenerator */
+	protected $URLGenerator;
+
+	/**
+	 * @param Factory $languageFactory
+	 * @param IURLGenerator $URLGenerator
+	 */
+	public function __construct(Factory $languageFactory, IURLGenerator $URLGenerator) {
+		$this->languageFactory = $languageFactory;
+		$this->URLGenerator = $URLGenerator;
+	}
+
+	protected function getL10N($languageCode = null) {
+		return $this->languageFactory->get(self::APP_NAME, $languageCode);
+	}
+
+	/**
+	 * The extension can return an array of additional notification types.
+	 * If no additional types are to be added false is to be returned
+	 *
+	 * @param string $languageCode
+	 * @return array|false
+	 */
+	public function getNotificationTypes($languageCode) {
+		$l = $this->getL10N($languageCode);
+
+		return [
+			self::TYPE_VIRUS_DETECTED => (string) $l->t('Infected files has been <strong>found</strong>'),
+		];
+	}
+
+	/**
+	 * For a given method additional types to be displayed in the settings can be returned.
+	 * In case no additional types are to be added false is to be returned.
+	 *
+	 * @param string $method
+	 * @return array|false
+	 */
+	public function getDefaultTypes($method) {
+		$defaultTypes = [
+			self::TYPE_VIRUS_DETECTED,
+		];
+
+		return $defaultTypes;
+	}
+
+	/**
+	 * A string naming the css class for the icon to be used can be returned.
+	 * If no icon is known for the given type false is to be returned.
+	 *
+	 * @param string $type
+	 * @return string|false
+	 */
+	public function getTypeIcon($type) {
+		switch ($type) {
+			case self::TYPE_VIRUS_DETECTED:
+				return 'icon-info';
```

Comment:
Ah, it is in core, I just saw the empty field on your screenshot, so I thought its not there.